packages/store-cache/src/utils.ts

```diff
@@ -188,15 +185,17 @@ function getScanArgsFromFilter<C extends StoreConfig, T extends keyof C["tables"
 
   // Transform scan args
   if (table) {
-    scanArgs.gte = key?.gte && recordToTuple(key.gte, getKeyOrder(config, table));
-    scanArgs.gt = key?.gt && recordToTuple(key.gt, getKeyOrder(config, table));
-    scanArgs.lte = key?.lte && recordToTuple(key.lte, getKeyOrder(config, table));
-    scanArgs.lt = key?.lt && recordToTuple(key.lt, getKeyOrder(config, table));
+    scanArgs.gte =
+      key?.gte && recordToTuple(key.gte as Record<string, StaticPrimitiveType>, getKeyOrder(config, table));
+    scanArgs.gt = key?.gt && recordToTuple(key.gt as Record<string, StaticPrimitiveType>, getKeyOrder(config, table));
+    scanArgs.lte =
+      key?.lte && recordToTuple(key.lte as Record<string, StaticPrimitiveType>, getKeyOrder(config, table));
+    scanArgs.lt = key?.lt && recordToTuple(key.lt as Record<string, StaticPrimitiveType>, getKeyOrder(config, table));
```

Comment:
TS didn't like using `StaticPrimitiveType` alongside `Key<C, T>` (I think because a table's key schema types aren't specific to just static ABI types?)

Reply:
ohh interesting, yeah i think you're right and fundamentally caused by the `keySchema` not explicitly being typed as `StaticAbiType` (we have validations for this but seems like somewhere the strong type gets lost). This is something we can revisit after #1077 is done.